src/box/lua/config/instance_config.lua

```diff
@@ -454,6 +454,66 @@ return schema.new('instance_config', schema.record({
                 end
             end,
         })),
+        tarantool = enterprise_edition(schema.record({
+            prefix = schema.scalar({
+                type = 'string',
+                validate = function(data, w)
+                    if not data:startswith('/') then
+                        w.error(('config.tarantool.prefix should be ' ..
+                            'a path alike value, got %q'):format(data))
+                    end
+                end,
+            }),
+            endpoints = schema.array({
+                items = schema.scalar({
+                    type = 'string',
+                }),
+                validate = function(data, w)
+                    if #data ~= 1 then
+                        w.error('only one endpoint in ' ..
+                            'config.tarantool.endpoints is supported now')
+                    end
+                end,
+            }),
+            username = schema.scalar({
+                type = 'string',
+            }),
+            password = schema.scalar({
+                type = 'string',
+            }),
+            connect_timeout = schema.scalar({
+                type = 'integer',
+            }),
+            ssl = schema.record({
+                key_file = schema.scalar({
+                    type = 'string',
+                }),
+                cert_file = schema.scalar({
+                    type = 'string',
+                }),
+                ca_file = schema.scalar({
+                    type = 'string',
+                }),
+                ciphers = schema.scalar({
+                    type = 'string',
+                }),
+                password = schema.scalar({
+                    type = 'string',
+                }),
+                password_file = schema.scalar({
+                    type = 'string',
+                })
+            }),
```

Comment:
Several points we discussed verbally with @askalt.

1. The idea was to attempt to make the options similar to `config.etcd`, but it seems that `net.box` options are too different. So, there is not consistency with `config.etcd` anyway and it worth to try to make the options consistent with `iproto.listen` and `iproto.advertise.{peer,sharding}`.
2. It is OK to split the tasks and implement the storage without TLS options support first and add them in a separate PR.
3. Timeout options (`connect_timeout` in this case) should accept a floating point number. The type in the schema should be changed from `integer` to `number`.
4. It is desirable to have meaningful defaults for timeout options, not infinity.

Reply:
I suggest to rename `connect_timeout` to `timeout` and use it as a connect timeout and a request timeout to make things easier.

Reply:
Updated.